packages/zoe/src/issuerTable.js

```diff
@@ -0,0 +1,126 @@
+// @ts-check
+
+import { assert } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+
+import makeWeakStore from '@agoric/weak-store';
+import makeAmountMath from '@agoric/ertp/src/amountMath';
+import { makeTable, makeValidateProperties } from './table';
+
+import '../exported';
+import './internal-types';
+
+/**
+ * @template K,V
+ * @typedef {import('@agoric/weak-store').WeakStore<K,V>} WeakStore
+ */
+
+// Issuer Table key: brand
+// Columns: issuer | amountMath | brand
+//
+// The IssuerTable is keyed by brand, but the Issuer is required in order for
+// getPromiseForIssuerRecord() to initialize the records. When
+// getPromiseForIssuerRecord is called and the record doesn't exist, it stores a
+// promise for the record in issuersInProgress, and then builds the record. It
+// updates the tables when done.
+//
+const makeIssuerTable = () => {
+  /**
+   * TODO: make sure this validate function protects against malicious
+   * misshapen objects rather than just a general check.
+   * @type {Validator<IssuerRecord>}
+   */
+  const validateSomewhat = makeValidateProperties([
+    'brand',
+    'issuer',
+    'amountMath',
+  ]);
+
+  /** @param {Table<IssuerRecord>} table */
+  const makeCustomMethods = table => {
+    /** @type {WeakStore<Issuer, Promise<IssuerRecord>>} */
+    const issuersInProgress = makeWeakStore('issuer');
+
+    /** @type {WeakStore<Issuer,Brand>} */
+    const issuerToBrand = makeWeakStore('issuer');
+
+    const registerIssuerRecord = issuerRecord => {
+      const { brand, issuer } = issuerRecord;
+      table.create(issuerRecord, brand);
+      issuerToBrand.init(issuer, brand);
+    };
+
+    // We can't be sure we can build the table entry soon enough that the first
+    // caller will get the actual data, so we start by saving a promise in the
+    // inProgress table, and once we have the Issuer, build the record, fill in
+    // the table, and resolve the promise.
+    /**
+     * @param {Issuer} issuer
+     */
+    function buildTableEntryAndPlaceHolder(issuer) {
+      // remote calls which immediately return a promise
+      const mathHelpersNameP = E(issuer).getMathHelpersName();
+      const brandP = E(issuer).getBrand();
+      const brandIssuerMatchP = E(brandP).isMyIssuer(issuer);
+
+      /**
+       * @type {[
+       *  PromiseLike<Brand>,
+       *  PromiseLike<'nat' | 'set' | 'strSet'>,
```

Comment:
```suggestion
       *  PromiseLike<MathHelpersName>,
```
after importing the type from ERTP